include/pros/link.hpp

```diff
@@ -0,0 +1,200 @@
+/**
+ * \file pros/link.hpp
+ *
+ * Contains prototypes for functions related to robot to robot communications.
+ *
+ * Visit https://pros.cs.purdue.edu/v5/tutorials/topical/link.html to learn
+ * more.
+ *
+ * This file should not be modified by users, since it gets replaced whenever
+ * a kernel upgrade occurs.
+ *
+ * Copyright (c) 2017-2021, Purdue University ACM SIGBots.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+#ifndef _PROS_LINK_HPP_
+#define _PROS_LINK_HPP_
+
+#include <cstdint>
+
+#include "pros/link.h"
+
+namespace pros {
+class Link {
+	private:
+	std::uint8_t _port;
+	char* _link_id;
```

Comment:
This is requiring _link_id to remain valid for the lifetime of Link. I'd recommend making a copy of it.

Reply:
You don't need _type and _link_id it looks like.

Reply:
storing a char* in c++ 🥴

Reply:
(i think you might as well make this a std::string, let the constructor handle conversion from the literal if necessary, and then pass the .c_str() down to the C API where needed. if only to deal with the lifetime issue that nathan brought up)

Reply:
yeah, I've talked to Ayush about this but I think he's busy and hasn't been able to fix it atm.